src/Cms/ModelWithContent.php

```diff
@@ -492,8 +492,18 @@ protected function saveTranslation(array $data = null, string $languageCode = nu
             throw new InvalidArgumentException('Invalid language: ' . $languageCode);
         }
 
+        // get the content to store
+        $content = $translation->update($data, $overwrite)->content();
```

Comment:
Is it needed to run the loop over all fields (even the ones that won't be modified by the current save operation)? It means that existing fields that have become untranslatable in the meantime will be kicked out. It's probably useful to do that, but we should be sure that this is what we want. Besides that the changes LGTM.

Reply:
I think it's better to do it that way and not to be too smart about the fields, which should be skipped or not.